File: events/joined.js
import dotenv from 'dotenv';
import { withBlacklistEvent } from './blacklist.js';
dotenv.config();

export function register(app) {
  app.event('member_joined_channel', withBlacklistEvent('member_joined', async ({ event, say, logger }) => {
    logger.info(event);
    const user = event.user;
    const channel = event.channel;

    try {
      const botUserId = (await app.client.auth.test()).user_id;

      const convoInfo = await app.client.conversations.info({ channel });
      const channelInfo = convoInfo.channel || {};
      const managerId = channelInfo.creator || event.inviter || botUserId; //if doesnt work then fallback to this

      await say({
      channel,
      text: `Hi <@${user}>, welcome to the land where <@${managerId}> does stupid shit`
      });
    } catch (err) {
      logger.error('Failed to post welcome message:', err);

      await say({
        channel,
        text: `Hi <@{user}>, welcome to the land where my creator does stupid shit`
      })
    }
  }));
}

File: events/joined.py
from dotenv import load_dotenv
import os
import logging
from events.blacklist import blockIfBlacklisted

load_dotenv()
logger = logging.getLogger(__name__)

def register(app):
    @app.event("member_joined_channel")
    @blockIfBlacklisted
    def handleMemberJoinedChannelEvents(event, say, logger, body):
        user = event["user"]
        channel = event["channel"]
        try:
            message = f'Hi <@{user}>, welcome to the land where <@{os.getenv("SLACK_USER_ID")}> writes shit code'
            say(channel=channel, text=message)

            msg = f"Sent welcome message to <@{user}> in channel {channel}"
            logger.info(f"[joined] {msg}")

        except Exception as e:
            errorMsg = f"Failed to send welcome message to <@{user}> in channel {channel}: {e}"
            logger.error(f"[joined] {errorMsg}")